pyEpiabm/pyEpiabm/sweep/assign_household_ages_sweep.py

```diff
@@ -0,0 +1,394 @@
+#
+
+import random
+from scipy.stats import poisson
+import numpy as np
+
+from pyEpiabm.core import Person
+from pyEpiabm.core import Parameters
+from .abstract_sweep import AbstractSweep
+
+
+class AssignHouseholdAgesSweep(AbstractSweep):
+    """Class to assign ages to people in
+    households using the same age distribution
+    model as covid-sim.
+    """
+
+    def __init__(self):
+        """Call in variables from the parameters file.
+        """
+        self.age_params = Parameters.instance().household_age_params
+        self.age_proportions = Parameters.instance().age_proportions
+        self.num_age_groups = len(self.age_proportions)
+        self.age_group_width = 5
+
+    def one_person_household_age(self, person: Person):
+        """Method that assigns an age to the person
+        in a one person household. A random number is first drawn
+        that decides which set of conditions the person's age
+        should satisfy, e.g. if they are old or young and single etc. The
+        person's age is then repeatedly set until it makes realsitic sense
+        e.g they are not a child living alone.
+
+        Parameters
+        ----------
+
+        Person : Person
+            Instance of Person class
+
+        """
+
+        r = random.random()
+
+        # case where an elderly person lives alone
+        if r < self.age_params["one_pers_house_prob_old"]:
+            while True:
+                person.set_random_age()
+                break_ratio = ((person.age
+                                - self.age_params["no_child_pers_age"] + 1)
+                               / (self.age_params["old_pers_age"]
+                               - self.age_params["no_child_pers_age"]
+                               + 1))
+                if ((person.age >= self.age_params["no_child_pers_age"]) and
+                   (random.random() <= break_ratio)):
+                    break
+
+        # case where a young and single adult lives alone
+        elif ((self.age_params["one_pers_house_prob_young"] > 0) and
+              (r < (self.age_params["one_pers_house_prob_young"]
+               / (1 - self.age_params["one_pers_house_prob_old"])))):
+            while True:
+                person.set_random_age()
+                break_ratio = (
+                    1 - self.age_params["young_and_single_slope"]
+                    * ((person.age - self.age_params["min_adult_age"])
+                        / (self.age_params["young_and_single"]
+                           - self.age_params["min_adult_age"])))
+                if ((person.age <= self.age_params["young_and_single"]) and
+                   (person.age >= self.age_params["min_adult_age"]) and
+                   (random.random() <= break_ratio)):
+                    break
+
+        # case where it is just one adult living alone
+        else:
+            while True:
+                person.set_random_age()
+                if person.age >= self.age_params["min_adult_age"]:
```

Comment:
So this person could also be elderly, or young & single?

Reply:
Correct. Have updated the comment to make this more explicit.